t/helper/test-gvfs-protocol.c

```diff
@@ -0,0 +1,1191 @@
+	flags |= OBJECT_INFO_LOOKUP_REPLACE;
+
+	if (!req_data->slash_args.len ||
+	    get_oid_hex(req_data->slash_args.buf, &oid)) {
+		logerror("%s: invalid OID in GET gvfs/objects: '%s'",
+			 TR2_CAT, req_data->slash_args.buf);
+		return -1;
+	}
+
+	trace2_printf("%s: GET %s", TR2_CAT, oid_to_hex(&oid));
+
+	oi.typep = &type;
+	oi.sizep = &size;
+	oi.contentp = &content;
+
+	if (oid_object_info_extended(the_repository, &oid, &oi, flags)) {
+		logerror("%s: Could not find OID: '%s'",
+			 TR2_CAT, oid_to_hex(&oid));
+		return -1;
+	}
+
+	// TODO....
+	trace2_printf("%s: OBJECT type=%d len=%ld '%.40s'", TR2_CAT,
+		      type, size, (const char *)content);
+
+	return my_send_loose_object(&oi, &oid, 1);
+}
+
+static int do__gvfs_objects__post(struct my_request_data *req_data)
+{
+	return -1;
+}
+	
+#if 0
+/*
+ * Parse line 0 of the HTTP request.  We expect something of the form:
+ *
+ *    GET [<uri_base>]/<gvfs_api>[/<args>] HTTP/1.1
+ *    GET [<uri_base>]/<gvfs_api>[?<args>] HTTP/1.1
+ *
+ * where:
+ *    <uri_base> is the URI (excluding the protocol and hostname) used
+ *               by the client.
+ *    <gvfs_api> is the REST API named in the GVFS Protocol.  This is
+ *               of the form "gvfs/objects".
+ *    <args>     is any args to the above, such as "<oid>".
+ *
+ * For example, when the cache-server or main Git server is set to
+ * something like "http://localhost:9999/", we get:
+ *
+ *    GET /gvfs/objects/aaaaaaaaaabbbbbbbbbbccccccccccdddddddddd HTTP/1.1
+ *    POST /gvfs/objects HTTP/1.1
+ *
+ * We can use the base part of the URI to support both types of servers
+ * on one port.  For example, with "http://localhost:9999/repos/" and
+ * "http://localhost:9999/cache/", we get:
+ *
+ *    GET /repos/gvfs/objects/aaaaaaaaaabbbbbbbbbbccccccccccdddddddddd HTTP/1.1
+ *    GET /cache/gvfs/objects/aaaaaaaaaabbbbbbbbbbccccccccccdddddddddd HTTP/1.1
+ *
+ * Or we can let <uri_base> be a series of key/value pairs to drive
+ * different configurations or unit tests.  For example:
+ *
+ *    GET /k1/v1/k2/v2/gvfs/objects/aaaaaaaaaabbbbbbbbbbccccccccccdddddddddd HTTP/1.1
+ * 
+ * We only expect a limited set of URLs for testing purposes, so we
+ * do not bother with full URL escape sequence decoding here.
+ */    
+#endif
+
+/*
+ * Read the HTTP request up to the start of the optional message-body.
+ * We do this byte-by-byte because we have keep-alive turned on and
+ * cannot rely on an EOF.
+ *
+ * https://tools.ietf.org/html/rfc7230
+ * https://github.com/microsoft/VFSForGit/blob/master/Protocol.md
+ *
+ * Returns 1 on EOF
+ * Returns 0 when we have a valid request
+ * Returns -1 when we have a parser error.
+ *
+ * We cannot call die() here because our caller needs to properly
+ * respond to the client and/or close the socket before this
+ * child exits so that the client doesn't get a connection reset
+ * by peer error.
+ */
+static int my_request_data__read(struct my_request_data *req_data, int fd)
```

Comment:
Since this function is `static` already, it can have a much shorter name, as it is definitely not polluting the global namespace. Also, if you call the data structure `struct req` and the variable `req`, it will make for a much succincter read, without sacrificing any robustness.